apps/cowswap-frontend/src/modules/ordersTable/pure/ReceiptModal/FeeField.tsx

```diff
@@ -7,23 +7,42 @@ import * as styledEl from './styled'
 
 export type Props = { order: ParsedOrder }
 
+// TODO: get rid of this once https://github.com/cowprotocol/services/pull/3184 is complete
+const HAS_BACKEND_MIGRATED = false
+
+function getFeeToken(order: ParsedOrder) {
+  if (!HAS_BACKEND_MIGRATED) {
+    return order.inputToken
+  }
+
+  const { inputToken, outputToken } = order
+  const { executedFeeToken } = order.executionData
+
+  if (inputToken?.address.toLowerCase() === executedFeeToken?.toLowerCase()) {
+    return inputToken
+  }
+  if (outputToken?.address.toLowerCase() === executedFeeToken?.toLowerCase()) {
+    return outputToken
+  }
+  return undefined
+}
```

Comment:
I had already implemented it assuming it would be in the surplus, but that's not the case yet.